doc/source/data/training_ingest.rst

```diff
@@ -0,0 +1,525 @@
+
+.. testcode::
+    import ray
+
+     ds = ray.data.read_text(
+        "s3://anonymous@ray-example-data/sms_spam_collection_subset.txt"
+    )
+
+    # Randomize the blocks of this dataset.
+    ds = ds.randomize_block_order()
+
+    def train_loop_per_worker():
+        # Get an iterator to the dataset we passed in below.
+        it = session.get_dataset_shard("train")
+        for _ in range(2):
+            # Use a shuffle buffer size 10k elements.
+            for batch in it.iter_batches(
+                local_shuffle_buffer_size=10000, batch_size=128):
+                print("Do some training on batch", batch)
+
+    my_trainer = TorchTrainer(
+        train_loop_per_worker,
+        scaling_config=ScalingConfig(num_workers=2),
+        datasets={"train": ds},
+    )
+    my_trainer.fit()
+
+.. testoutput::
+    :hide:
+
+    ...
+
+
+If your model is very dependent on shuffle quality, you can do a full global shuffle by calling :meth:`random_shuffle <ray.data.Dataset.random_shuffle>` on your dataset.
+
+.. testcode::
+
+    import ray
+
+     ds = ray.data.read_text(
+        "s3://anonymous@ray-example-data/sms_spam_collection_subset.txt"
+    )
+
+    # Do a global shuffle of all rows in this dataset.
+    ds = ds.random_shuffle()
+
+For more information on how to optimize shuffling, and which approach to choose, see the :ref:`Optimize shuffling guide <optimizing_shuffles>`.
+
+Reproducibility
+~~~~~~~~~~~~~~~
+When developing or hyperparameter tuning models, it's important to have reproducibility during data ingest so that it does not affect model quality. Follow these 3 steps to enable reproducibility:
+
+**Step 1:** Enable deterministic execution in Ray Datasets by setting the `preserve_order` flag in the :class:`DataContext <ray.data.context.DataContext>`.
+
+.. testcode::
+    import ray
+    from ray.air import ScalingConfig, session
+    from ray.train.torch import TorchTrainer
+
+    # Preserve ordering in Ray Datasets.
+    ctx = ray.data.DataContext.get_current()
+    ctx.execution_options.preserve_order = True
+
+    ds = ray.data.read_text(
+        "s3://anonymous@ray-example-data/sms_spam_collection_subset.txt"
+    )
+    train_ds, val_ds = dataset.train_test_split(0.3)
+
+    def train_loop_per_worker():
+        # Get an iterator to the dataset we passed in below.
+        it = session.get_dataset_shard("train")
+        for _ in range(2):
+            for batch in it.iter_batches(batch_size=128):
+                print("Do some training on batch", batch)
+
+    my_trainer = TorchTrainer(
+        train_loop_per_worker,
+        scaling_config=ScalingConfig(num_workers=2),
+        datasets={"train": train_ds, "val": val_ds},
+    )
+    my_trainer.fit()
+
+**Step 2:** Set a seed for any shuffling operations: 
+
+* :meth:`random_shuffle <ray.data.Dataset.random_shuffle>`
```

Comment:
I'm wondering why this isn't a code example. Where does the user put these API calls?

Reply:
wherever they put it, they just need to make sure they specify the right arguments